pandas/tests/series/test_period.py

```diff
@@ -164,3 +164,10 @@ def test_end_time_timevalues(self, input_vals):
         result = s.dt.end_time
         expected = s.apply(lambda x: x.end_time)
         tm.assert_series_equal(result, expected)
+
+    def test_to_period(self):
+        series = pd.Series(['2001'], dtype='datetime64[ns]')
+        assert series.dt.to_period('D').dtype == 'Period[D]'
```

Comment:
Construct the expected series and compare using `tm.assert_series.equals(result, expected)` so we can test that more than the dtype is correct.